Int: Redo internal directory structure

SebastianMorawiec opened the pull request:
# The problem

We want to split SDK into smaller packages so we can desync our client and server releaseses

# This PR's solution

Prepere interim dir sturcture.
https://zapatacomputing.atlassian.net/browse/ORQSDK-1029

Move everything ray-related into `runtime/` dir
Move everything shared between runtime and clients into `shared` dir
Create shims for:
- public API (dremio, mlflow, secrets)
- backward compatibility (unpickling old WF results)

Make sure that shared can be done as stand-alone package.
Make sure that runtime only imports things from shared

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

SebastianMorawiec committed:
fix tests

SebastianMorawiec committed:
fix perf test and style

SebastianMorawiec committed:
fix style and pyproject toml

SebastianMorawiec committed:
fix some tests and style

SebastianMorawiec committed:
fix docs

SebastianMorawiec committed:
add shims, move few files to shared

SebastianMorawiec committed:
Merge remote-tracking branch 'origin/main' into Morawiec/dir_restruct

SebastianMorawiec committed:
fix some tests

SebastianMorawiec committed:
fix more tests

SebastianMorawiec committed:
fix all tests

SebastianMorawiec committed:
move ray envs to runtime dir

SebastianMorawiec committed:
move serde to shared

SebastianMorawiec committed:
fix style

SebastianMorawiec committed:
style fix

SebastianMorawiec committed:
move more dependencies around

SebastianMorawiec committed:
and even more files moved around....

SebastianMorawiec committed:
move logs struct to shared

SebastianMorawiec committed:
move spaces api to shared

SebastianMorawiec committed:
fix shims

SebastianMorawiec committed:
fix perf tests

alexjuda commented on src/orquestra/sdk/shared/_spaces/_structs.py:
```diff
@@ -3,7 +3,7 @@
 from dataclasses import dataclass
 
-from orquestra.sdk.schema.workflow_run import ProjectId, WorkspaceId
+from orquestra.sdk.shared.schema.workflow_run import ProjectId, WorkspaceId
```
**thought**: if we agree to only use relative imports within the same to-be-wheel, we'll have to change this and similar imports, too.

alexjuda commented on src/orquestra/sdk/shared/kubernetes/__init__.py:
**issue**: missing shim for backwards compatibility with our public API

alexjuda commented on tests/runtime/ray/data/python_package/python_package_dependent_workflow.json:
```diff
@@ -3,7 +3,7 @@
     "module": "__main__",
     "function_name": "wf",
-    "file_path": "original_workflow.py",
+    "file_path": "tests/runtime/ray/data/python_package/original_workflow.py",
```
**thought**: this change is weird, do you know why it was needed?

SebastianMorawiec replied:
I just regenerated this whole json. This was only side-effect of that which doesnt matter really.

alexjuda commented on src/orquestra/sdk/_client/_base/_traversal.py:
```diff
@@ -17,11 +17,15 @@
+)
+from orquestra.sdk.shared.schema import ir, responses
+
+from ...shared import _exec_ctx, serde
```
Reasoning: this is an import across boundaries of the to-be-wheels -> it will need to be an absolute import

alexjuda replied:
Changed in https://github.com/zapata-engineering/orquestra-sdk/pull/387

SebastianMorawiec commented on src/orquestra/sdk/runtime/__init__.py:
Good question: I followed the way how I understood the https://zapatacomputing.atlassian.net/wiki/spaces/ORQSRUN/pages/851280058/Client-Server+Decoupling+Jan+2024#Interim-Solution. I Didnt even give a second thought about it that it doesnt make sense, but thats how it was presented there 😅

alexjuda commented on src/orquestra/sdk/__init__.py:
```diff
@@ -29,14 +31,18 @@
+    WorkflowTemplate,
+    workflow,
+)
+from .shared._spaces._structs import Project, ProjectRef, Workspace
```
Okay, we did mention it in the [design doc](https://zapatacomputing.atlassian.net/wiki/spaces/ORQSRUN/pages/851280058/Client-Server+Decoupling+Jan+2024#Interim-Solution)

> Files under src/orquestra/sdk/runtime/:
> * Can import public symbols from shared.
> * Can import any file from under runtime (the same “future package”).
> * Must not import files from _client
>
> src/orquestra/sdk/__init__.py and files under _client:
> * Can import public symbols from shared.
> * Can import public symbols from runtime.
> * Can import any file from under _client (the same “future package”).

SebastianMorawiec replied:
Agreed to put it on separate PR offline

alexjuda commented on src/orquestra/sdk/_client/_base/_api/_config.py:
```diff
@@ -9,17 +9,17 @@
-
-from ...exceptions import ConfigFileNotFoundError, ConfigNameNotFoundError
-from ...schema.configs import (
+from orquestra.sdk._client._base._factory import build_runtime_from_config
```
https://github.com/zapata-engineering/orquestra-sdk/pull/387

alexjuda committed:
int: fix relative/absolute imports across to-be-wheels (#387)

# The problem

Imports weren't following the conventions established in the [design
doc](https://zapatacomputing.atlassian.net/wiki/spaces/ORQSRUN/pages/851280058/Client-Server+Decoupling+Jan+2024#Interim-Solution).

# This PR's solution

* Uses relative imports when importing only within the same "wheel".
* Uses absolute imports when importing only across different "wheels".

# Checklist

- [x] Tests have been added for new features/changed behavior (if no new
features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a
user-readable description of the changes (if the change isn't visible to
the user in any way, check the box).
- [x] The PR's title is prefixed with
`<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket,
check the box).

SebastianMorawiec committed:
move exceptions to shared dir

SebastianMorawiec commented on src/orquestra/sdk/_client/__init__.py:
agreed to do on separate PR

SebastianMorawiec committed:
Style fix

SebastianMorawiec committed:
ir shim

SebastianMorawiec committed:
fix responses shim

SebastianMorawiec committed:
fix all shims in schema

SebastianMorawiec committed:
fix exceptions shims

SebastianMorawiec committed:
kubernetes shim

SebastianMorawiec committed:
packaging shim

SebastianMorawiec committed:
fix _ray shim

SebastianMorawiec committed:
rename runtime dir to _runtime

SebastianMorawiec committed:
rename shared to _shared dir

alexjuda reviewed (approved):
Thanks for getting through all this!

SebastianMorawiec committed:
Style fix

SebastianMorawiec committed:
fix invalid imports

SebastianMorawiec committed:
fix style and tests

SebastianMorawiec committed:
fix style and changelog add

SebastianMorawiec committed:
fix remaining tests
